### ISSUE TITLE
Computed ordinalValue is unexpected

### ISSUE THREAD

**Issue by jellynoone** [superlistapp/super_editor#1938]
**Package Version**
`main`

**To Reproduce**
1. Run the example application (test on web)
2. Scroll down to `Explore the toolkit 🔎`
3. Select last unordered list item under that section
4. Change the paragraph to "ordered list item"
5. Observe the `ordinalValue` being `5.`

**Actual behavior**
When an ordered list item is directly preceded by multiple unordered list items, its count is as if those unordered list items were ordered.

**Expected behavior**
I'd expect that the ordinalValue is computed based on previous list items that are also of "ordered" type not just all preceding list items with the same indent.
In other words a list such as:
```md
* item
* item
1. item
```
should translate to the above, rather than:
```md
* item
* item
3. item
```

**Platform**
Any

**Additional**
The problem seems to be here: https://github.com/superlistapp/super_editor/blob/661ecb84b4f79bd3f81c1bfd148b36ab7be6993b/super_editor/lib/src/default_editor/list_items.dart#L113-L123
Here computing the ordinal values doesn't take into consideration the type of the preceded ListItem.
